### PR TITLE
Allow customizing building from source

### PR DESCRIPTION
**Pull request by elfosardo**
This patch introduces some variables and improvements that can help
customizing the ironic-image when building it from source.
In particular, it adds the option to install ironic, ironic-inspector,
or the sushy library from a gerrit refs or a commit hash.

### CODE REVIEW

File: ironic-source-list.j2 [hunk 1]
```diff
@@ -0,0 +1,18 @@
+crudini
+gunicorn
+{% if env.IRONIC_SOURCE_HASH %}
+ironic @ git+https://opendev.org/openstack/ironic@{{ env.IRONIC_SOURCE_HASH }}
+{% elif env.IRONIC_SOURCE_GERRIT %}
+ironic @ git+https://opendev.org/openstack/ironic@{{ env.IRONIC_SOURCE_GERRIT }}
```
Comment (dtantsur) [inline ironic-source-list.j2:6]: Looks identical to the previous case, maybe use the same option?

File: ironic-source-list.j2 [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+{% else %}
+ironic @ git+https://opendev.org/openstack/ironic
+{% endif %}
+ironic-inspector @ git+https://opendev.org/openstack/ironic-inspector
```
Comment (dtantsur) [inline ironic-source-list.j2:10]: Will likely need the same treatment.

File: ironic-source-list.j2 [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+ironic-prometheus-exporter
+proliantutils
+PyMySQL>=0.8.0
+python-dracclient
+python-ibmcclient
+python-scciclient
+sushy
```
Comment (dtantsur) [inline ironic-source-list.j2:17]: Ideally here as well

File: ironic-source-list.j2 (hunk 1 above)
Comment (elfosardo) [reply ironic-source-list.j2:6]: yes, I will use the same option
it will be one option for each custom source

File: ironic-source-list.j2 (hunk 2 above)
Comment (elfosardo) [reply ironic-source-list.j2:10]: yeah, in the end I'll try to cover as much packages/libraries as possible

**Comment by elfosardo** [comment]
weird resolution issue for https://raw.githubusercontent.com

**Pull request merged**